<launch>
    <!-- Launch this file through the gui in living_at_diag/scripts/launcher.py -->
    
    <arg name="map_name"  default="Basement"/>
    <arg name="robot_name" default="bigRodent"/>
    <arg name="init_x" default="16.46" />
    <arg name="init_y" default="12.03" />
    <arg name="init_a" default="-3.099" />
    <arg name="init_cov" default="0.25" />
    <arg name="robotPort" default="/dev/ttyUSB0"/>
    <arg name="laserPort" default="/dev/ttyACM0"/>
    <arg name="joyPort" default="/dev/ttyACM3"/>        
    <!--arg name="home_x" default="8.0" />
    <arg name="home_y" default="2.0" />
    <arg name="home_th" default="0.0" />
    
    <param name="/home_x" value="$(arg home_x)"/> 
    <param name="/home_y" value="$(arg home_y)"/> 
    <param name="/home_th" value="$(arg home_th)"/--> 
    <param name="robot_name" value="$(arg robot_name)"/>
    
    <!--LOAD MAP-->
    <include file="$(find living_at_diag)/launch/map.launch">
      <arg name="map_name" value="$(arg map_name)" />
    </include>
    
    <group ns="$(arg robot_name)">
	<param name="robot_name" value="$(arg robot_name)" />
	<remap from="cmd_vel" to ="desired_cmd_vel" />

	<!--LOAD AMCL-->
	<node pkg="amcl" type="amcl" name="amcl" respawn="true">
                
                <remap from="static_map" to="/static_map" />
                
                <!-- Default parameters -->
                <rosparam file="$(find sapienzbot_setup)/config/localization/amcl.yaml" command="load" />
                
                <!-- Override AMCL Frame Params to include prefix -->
                <param name="base_frame_id" value="/$(arg robot_name)/base_link"/>
                <param name="odom_frame_id" value="/$(arg robot_name)/odom"/>
                <param name="global_frame_id" value="/map"/> <!--common map frame for all robots -->

                <param name="initial_pose_x" value="$(arg init_x)" />
                <param name="initial_pose_y" value="$(arg init_y)" />
                <param name="initial_pose_a" value="$(arg init_a)" />
                <param name="initial_cov_xx" value="0.25" />
                <param name="initial_cov_yy" value="0.25" />
                <param name="initial_cov_aa" value="0.25" />

		
        </node>

	<!--LOAD MOVE BASE-->
	<node pkg="move_base" type="move_base" respawn="false" name="move_base_node" output="screen">
                <remap from="map" to="/map" />
                <remap from="odom" to="odom" />
                
                <rosparam file="$(find sapienzbot_setup)/config/navigation/move_base_params.yaml" command="load" />
                <rosparam file="$(find sapienzbot_setup)/config/navigation/costmap_common_params.yaml" command="load" ns="global_costmap" />
                <rosparam file="$(find sapienzbot_setup)/config/navigation/costmap_common_params.yaml" command="load" ns="local_costmap" />
                <rosparam file="$(find sapienzbot_setup)/config/navigation/local_costmap_params.yaml" command="load" />
                <rosparam file="$(find sapienzbot_setup)/config/navigation/global_costmap_params.yaml" command="load" />
                <rosparam file="$(find sapienzbot_setup)/config/navigation/base_local_planner_params.yaml" command="load" />
                
                <!-- Override MOVE_BASE Frame Params to include prefix -->
                <param name="global_costmap/laser_scan_sensor/sensor_frame" value="/$(arg robot_name)/laser"/>
                <param name="global_costmap/laser_scan_sensor/topic" value="/$(arg robot_name)/scan"/>
                <param name="global_costmap/robot_base_frame" value="/$(arg robot_name)/base_link"/>   
                <param name="local_costmap/global_frame" value="/$(arg robot_name)/odom"/>
                <param name="local_costmap/laser_scan_sensor/sensor_frame" value="/$(arg robot_name)/laser"/>
                <param name="local_costmap/laser_scan_sensor/topic" value="/$(arg robot_name)/scan"/>
                <param name="local_costmap/robot_base_frame" value="/$(arg robot_name)/base_link"/>
                
        </node>
    
	<!--LOAD CAPYBARA-->
        <node name="capybara_name" pkg="capygroovy" type="shinyCapybara.py" output="screen">
		<param name="serialPort" value="$(arg robotPort)"/>
	</node>

	<!--LOAD LASER-->
	<node name="hokuyo_node" pkg="hokuyo_node" type="hokuyo_node">
		<param name="frame_id" value="/$(arg robot_name)/laser"/>
		<param name="port" value="$(arg laserPort)" />
	</node>	

	<!--LOAD UNA COSA CHE NON MI PIACE-->
	<node pkg="laser_analysis" type="laserobstacle" args=""  name="laserobstacle" output="screen">
            <param name="~laser_frame_id" value="$(arg robot_name)/laser" />
        </node>

        <!--LOAD BROADCASTERS-->
	<node name="link_broadcaster_0" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 /$(arg robot_name)/base_footprint /$(arg robot_name)/base_link 1" />
	<node name="link_broadcaster_1" pkg="tf" type="static_transform_publisher" args="0.2 0 0.24 0 0 0 /$(arg robot_name)/base_link /$(arg robot_name)/laser 1" />
        <node name="link_broadcaster_2" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 /$(arg robot_name)/base_link /$(arg robot_name)/base_frame 1" />
        <node name="link_broadcaster_3" pkg="tf" type="static_transform_publisher" args="-0.01 0 0.05 0 0 0 /$(arg robot_name)/laser /$(arg robot_name)/top_link 1" />
        <node name="link_broadcaster_4" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 /$(arg robot_name)/top_link /camera_link 1" />

	
	<!--node pkg="emergency_stop" name="emerg" type="emergency_stop" args="-gui" output="screen"/-->

    </group> 

</launch>
